src/core/gltf2importer/animationparser.cpp

```diff
@@ -249,6 +262,14 @@ Qt3DAnimation::QChannel AnimationParser::animationChannelDataFromBuffer(const QS
     std::vector<float> valueStamps;
     std::tie(nbComponents, valueStamps) = animationChannelDataFromData(sampler);
 
+    // Check nbComponent and type are what is expected for a given channel
+    const quint8 expectedComponents = expectedComponentsCountForChannel(channelName);
+
```

Comment:
and if warning is printed above, have an early return on 0 result (not printing anything else)

Reply:
done